From: Dimitri John Ledkov <xnox@ubuntu.com>
Date: Wed, 12 Sep 2018 19:51:23 +0100
Subject: core: execute environment_generators with manager's environment

(cherry picked from commit ea368f0bd2b77bbc67eab42471b470582f0bd6bc)
---
 src/core/manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/manager.c b/src/core/manager.c
index 7ccef8e..3afa39a 100644
--- a/src/core/manager.c
+++ b/src/core/manager.c
@@ -3406,7 +3406,7 @@ static int manager_run_environment_generators(Manager *m) {
         if (!generator_path_any(paths))
                 return 0;
 
-        return execute_directories(paths, DEFAULT_TIMEOUT_USEC, gather_environment, args, NULL, NULL);
+        return execute_directories(paths, DEFAULT_TIMEOUT_USEC, gather_environment, args, NULL, m->environment);
 }
 
 static int manager_run_generators(Manager *m) {
